### ISSUE TITLE
package installed as 0.0.0

### ISSUE THREAD

**Issue by jaraco** [pypa/setuptools-scm#22]
Using easy_install to install setuptools_scm, it installs as version 0.0.0:

```
C:\Users\jaraco\scm> $env:PYTHONPATH='.'
C:\Users\jaraco\scm> easy_install -d . setuptools_scm
﻿Creating c:\users\jaraco\scm\site.py
Searching for setuptools-scm
Reading https://pypi.python.org/simple/setuptools_scm/
﻿C:\Program Files\Python34\lib\site-packages\setuptools-12.2-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-0.26)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
C:\Program Files\Python34\lib\site-packages\setuptools-12.2-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.0.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
C:\Program Files\Python34\lib\site-packages\setuptools-12.2-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.1.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
Best match: setuptools-scm 1.1.0
Downloading https://pypi.python.org/packages/source/s/setuptools_scm/setuptools-scm-1.1.0.tar.gz#md5=a4a06bc7c862f63029793085734adcfb
Processing setuptools-scm-1.1.0.tar.gz
Writing C:\Users\jaraco\AppData\Local\Temp\easy_install-xeq_tah8\setuptools-scm-1.1.0\setup.cfg
Running setuptools-scm-1.1.0\setup.py -q bdist_egg --dist-dir C:\Users\jaraco\AppData\Local\Temp\easy_install-xeq_tah8\setuptools-scm-1.1.0\egg-dist-tmp-yzff7icr
C:\Program Files\Python34\lib\distutils\dist.py:260: UserWarning: Unknown distribution option: 'use_scm_version'
  warnings.warn(msg)
zip_safe flag not set; analyzing archive contents...
﻿Copying setuptools_scm-0.0.0-py3.4.egg to c:\users\jaraco\scm
Adding setuptools-scm 0.0.0 to easy-install.pth file

Installed c:\users\jaraco\scm\setuptools_scm-0.0.0-py3.4.egg
Processing dependencies for setuptools-scm
Finished processing dependencies for setuptools-scm
C:\Users\jaraco\scm> python
Python 3.4.2 (v3.4.2:ab2c023a9432, Oct  6 2014, 22:16:31) [MSC v.1600 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
>>> pkg_resources.require('setuptools_scm')[0].version
'0.0.0'
```

It should install as the advertised version.

---
- Bitbucket: https://bitbucket.org/pypa/setuptools_scm/issue/22
- Originally reported by: Jason R. Coombs
- Originally created at: 2015-02-25T17:12:40.765

**Comment by jaraco** [comment]
print the note when using setup.py in a fresh checkout, fixes #22

→ <<cset 5621ec8f0a76>>

---

Original comment by: Ronny Pfannschmidt

**Comment by jaraco** [comment]
I'm afraid I don't understand. How does printing the note in the setup.py file fix the PYPI hosted tarballs to get the correct version number? Is that just a reminder for the maintainer for the next release?

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
I still have this problem:

```
C:\Users\jaraco\scm> $env:PYTHONPATH='.'
C:\Users\jaraco\scm> easy_install -d . setuptools_scm
﻿Creating c:\users\jaraco\scm\site.py
Searching for setuptools-scm
Reading https://pypi.python.org/simple/setuptools_scm/
﻿c:\python\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-0.26)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
c:\python\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.0.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
c:\python\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.1.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
c:\python\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.2.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
Best match: setuptools-scm 1.2.0
Downloading https://pypi.python.org/packages/source/s/setuptools_scm/setuptools-scm-1.2.0.tar.gz#md5=251730e2ca3ff05e6e1fcd4ba368e6a8
Processing setuptools-scm-1.2.0.tar.gz
Writing C:\Users\jaraco\AppData\Local\Temp\easy_install-ally0u56\setuptools-scm-1.2.0\setup.cfg
Running setuptools-scm-1.2.0\setup.py -q bdist_egg --dist-dir C:\Users\jaraco\AppData\Local\Temp\easy_install-ally0u56\setuptools-scm-1.2.0\egg-dist-tmp-bteox7a4
c:\python\lib\distutils\dist.py:260: UserWarning: Unknown distribution option: 'use_scm_version'
  warnings.warn(msg)
zip_safe flag not set; analyzing archive contents...
﻿Copying setuptools_scm-0.0.0-py3.4.egg to c:\users\jaraco\scm
Adding setuptools-scm 0.0.0 to easy-install.pth file

Installed c:\users\jaraco\scm\setuptools_scm-0.0.0-py3.4.egg
Processing dependencies for setuptools-scm
Finished processing dependencies for setuptools-scm
```

This behavior means that every time I install the package (or every time setuptools installs it as part of a setup_requires), the apparent version of the package is 0.0.0. I'm unable to infer what version is being used.

Furthermore, if the setup_requires requirement stipulates anything that doesn't include 0.0.0, invocation will fail. Consider this minimal setup.py:

```
import setuptools

setuptools.setup(
    name='test_pkg',
    use_scm_version=True,
    setup_requires=[
        'setuptools_scm>=1.0',
    ],
)
```

Invoke it for sdist and it yields this output:

```
C:\Users\jaraco\scm> ./setup sdist
﻿C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-0.26)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.0.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.1.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py:2512: PEP440Warning: 'setuptools (scm-1.2.0)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
C:\Program Files\Python34\lib\distutils\dist.py:260: UserWarning: Unknown distribution option: 'use_scm_version'
  warnings.warn(msg)
zip_safe flag not set; analyzing archive contents...
﻿
Installed c:\users\jaraco\scm\.eggs\setuptools_scm-0.0.0-py3.4.egg
Traceback (most recent call last):
  File "C:\Users\jaraco\scm\setup.py", line 7, in <module>
    'setuptools_scm>=1.0',
  File "C:\Program Files\Python34\lib\distutils\core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\setuptools\dist.py", line 265, in __init__
  File "C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\setuptools\dist.py", line 310, in fetch_build_eggs
  File "C:\Program Files\Python34\lib\site-packages\setuptools-14.0-py3.4.egg\pkg_resources\__init__.py", line 807, in resolve
pkg_resources.DistributionNotFound: setuptools-scm>=1.0
```

It's downloaded setuptools_scm 1.2, but it's installed as 0.0:

```
C:\Users\jaraco\scm> dir .eggs


    Directory: C:\Users\jaraco\scm\.eggs


Mode                LastWriteTime     Length Name
----                -------------     ------ ----
-a---       13-Mar-2015     15:36        217 README.txt
-a---       13-Mar-2015     15:36      16978 setuptools_scm-0.0.0-py3.4.egg

```

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
I had this issue in hgtools, which is why [hgtools couldn't use itself naturally](https://bitbucket.org/jaraco/hgtools/src/8734a0ddb337d139a0f8b777c463aac7742465af/setup.py?at=default#cl-21). 

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
this seems to be a bug in my release process - as soon as the egg-info exist it can extract the data i will create some extra testing to fix it up

---

Original comment by: Ronny Pfannschmidt

**Comment by jaraco** [comment]
cant replicate on python2

---

Original comment by: Ronny Pfannschmidt

**Comment by jaraco** [comment]
seems like setuptools easyinstall sandboxing breaks the world

---

Original comment by: Ronny Pfannschmidt

**Comment by jaraco** [comment]
extend sdist command, replace version in setup.py of the sdist, fixes issue #22

→ <<cset b9902b92c9f1>>

---

Original comment by: Ronny Pfannschmidt
